presto-cache/src/main/java/com/facebook/presto/cache/UnboundedCacheManager.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cache;
+
+import com.facebook.airlift.log.Logger;
+import io.airlift.slice.Slice;
+import org.apache.hadoop.fs.Path;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+
+/**
+ * A naive and self-contained caching system that does not consider ranges of a file:
+ * - It simply uses offset and length as the key.
+ * - It does not consider memory utilization; flushes every requested data to disk.
+ * - It does not expire cache on disk.
+ *
+ * Experiment or test only. Do not use in production.
+ */
+public class UnboundedCacheManager
+        implements CacheManager
+{
+    private static final Logger log = Logger.get(UnboundedCacheManager.class);
+
+    private static final String EXTENSION = ".cache";
+
+    private final ScheduledExecutorService executor = newScheduledThreadPool(5);
```

Comment:
nit: Add thread name to the pool. Also ideally we should shutdown the thread pool properly.

Reply:
I don't think we need a "scheduled" thread pool here? A fix size thread pool will work for our use case here as there is no jobs that need to be run periodically.